models.ts

```diff
@@ -9,10 +9,50 @@ class Peli {
 
 class PelisCollection {
   getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+    return jsonfile.readFile("./pelis.json").then((file) => {
+      return file;
+    });
+  }
+  getById(id: number): Promise<any> {
+    return this.getAll().then((file) => {
+      return find(file, { id: id });
+    });
+  }
+  search(options: any): Promise<any> {
+    return this.getAll().then((file) => {
+      let arrayResultado = file;
+      if (options.title) {
+        arrayResultado = arrayResultado.filter((pelicula) => {
+          return pelicula.title.includes(options.title);
+        });
+      }
+      if (options.tag) {
+        arrayResultado = arrayResultado.filter((pelicula) => {
+          return pelicula.tags.find((tag) => {
+            return tag.includes(options.tag);
+          });
+        });
+      }
+      return arrayResultado;
+    });
+  }
+  add(peli: Peli): Promise<any> {
+    return this.getAll().then((file) => {
+      if (!find(file, { id: peli.id }) && peli.title && peli.tags) {
+        file.push(peli);
```

Comment:
En toda la primera parte podrías reutilizar this.getById para que esta funcion haga solo lo suyo. Pero es una mejora.